autoload/neomake/makers/ft/cs.vim

```diff
@@ -10,3 +10,34 @@ function! neomake#makers#ft#cs#mcs()
         \ 'errorformat': '%f(%l\,%c): %trror %m',
         \ }
 endfunction
+
+function! neomake#makers#ft#cs#msbuild()
+    return {
+        \ 'exe': 'msbuild',
+        \ 'args': [ '-nologo', '-v:q', '-property:GenerateFullPaths=true', s:FindSLN() ],
+        \ 'errorformat': '\ %#%f(%l\\\,%c):\ %m',
+        \ 'append_file' : 0,
+        \ }
+endfunction
+
+" FUNCTION find *.sln file going up directory
+function! s:FindSLN()
```

Comment:
We should have a generic utils function for this, probably based on what Syntastic has already.